CMakeLists.txt

```diff
@@ -129,8 +129,8 @@ cuda_add_library(gdf SHARED
     #src/windowedops.cu
 )
 
-
-target_link_libraries(gdf arrow)
+# boost_regex may be necessary only when arrow==0.9.0
+target_link_libraries(gdf arrow boost_regex)
```

Comment:
Is this different with 0.10.0?

Reply:
Yes, boost_regex is not needed for arrow 0.10.0 (tested). This issue is related to https://issues.apache.org/jira/browse/ARROW-2903 , in summary, linking against boost_regex is required for arrow versions `>=0.8` and `<=0.9.0`.

Note that the change in
https://github.com/gpuopenanalytics/libgdf/pull/89/files#diff-281a0db11f8c4d7c230f268619adb8bbR57
has the same origin.

Modifying these parts depends on the question, do we want to support arrow versions from 0.7.1 upto to the latest (0.10 currently) or can we drop support for some intermediate versions (such as 0.8, 0.9)?

Reply:
Can we make a conditional check for if Arrow is specified as one of those versions we add boost_regex, otherwise we don't? I think supporting 0.8 - 0.9.0 would be good, but we should avoid including boost_regex unnecessarily.

Reply:
Yes, we can. I will do it.